src/page/product-list.page.ts

```diff
@@ -1,14 +1,23 @@
-import { $, ElementFinder, browser, ExpectedConditions } from 'protractor';
+import { $$, ElementFinder, browser, ExpectedConditions, ElementArrayFinder } from 'protractor';
 
 export class ProductListPage {
-  private addToCartButton: ElementFinder;
+  private products: ElementArrayFinder;
 
   constructor () {
-    this.addToCartButton = $('ul.product_list a[data-id-product="1"][title="Add to cart"]');
+    this.products = $$('.product-container');
```

Comment:
Can you explain to me what `$$` does? is this a shortcut for something?

Reply:
It returns an array of elements (More precisely, an ElementArrayFinder) that are located with the css selector passed as an argument. It is different from a single $ because that one is supposed to locate only one element (returns ElementFinder).

It is a shortcut for element.all(by.css(...)), I found it here: https://www.protractortest.org/#/api?view=ElementArrayFinder